services/app/src/contexts/ChessBoardContext.jsx

```diff
@@ -135,15 +141,27 @@ const ChessBoardContextProvider = ({ children }) => {
     }
   }, [movedBy]);
 
+  const finishMove = (turn) => {
+    ws.send({
+      type: 'FigureMoved',
+      payload: { ...turn, player: userContext.username, id: currentPlayerId },
+    });
+  };
+
   const placeItemOnBoard = (row, tile, figure) => {
-    if (!finished && selectedItem) {
-      const copyOfBoard = [...board];
+    if (selectedItem === PieceName.SKIP) {
+      return;
+    }
+
+    if (selectedItem) {
+      const copyOfBoard = [...defaultBoard];
       const figureName = figure || selectedItem;
       if (lastTurn) {
         copyOfBoard[lastTurn.row][lastTurn.tile].items.length = 0;
       }
-      copyOfBoard[row][tile].items.push({ figure: figureName, score: getPieceScore(figureName), player: userContext.username, id: currentPlayerId });
       setLastTurn({ row, tile, figure: selectedItem });
```

Comment:
lastTurn seems to be very closely related to finishMove and setting item on the board.
Would it be possible to extract lastMove from the board using useMemo instead of setting it every time manually?